packages/components/src/internal/components/samples/SampleAssayDetail.tsx

```diff
@@ -159,50 +270,173 @@ const SampleAssayDetailBodyImpl: FC<SampleAssayDetailBodyProps & InjectedQueryMo
+
+                return assay.hasLookup(sampleModel.queryInfo.schemaQuery);
+            })
             .sort(naturalSortByProperty('name'))
             .reduce((_configs, assay) => {
                 const _queryConfig = createQueryConfigFilteredBySample(
                     assay,
-                    sampleId,
-                    Filter.Types.EQUAL,
-                    (fieldKey, sampleId) => `${fieldKey} = ${sampleId}`,
+                    sampleIds && sampleIds.length > 0 ? sampleIds : [-1],
+                    Filter.Types.IN,
+                    (fieldKey, sampleIds) => `${fieldKey} IN (${sampleIds.join(',')})`,
                     false,
-                    true
+                    !isSourceSampleAssayGrid
                 );
 
                 if (_queryConfig) {
-                    const modelId = `assay-detail:${assay.id}:${sampleId}`;
+                    const modelId = `assay-detail:${assay.id}:${sampleId ?? sourceId + '-source'}`;
                     _configs[modelId] = _queryConfig;
                     _tabOrder.push(modelId);
                 }
 
                 return _configs;
             }, {});
 
+        // keep tab when "all" view has data, but filtered view is blank
+        if (showAliquotViewSelector && activeSampleAliquotType && activeSampleAliquotType != ALIQUOT_FILTER_MODE.all) {
+            const unfilteredConfigs = assayModel.definitions
+                .slice() // need to make a copy of the array before sorting
+                .filter(assay => {
+                    if (isSourceSampleAssayGrid) return true;
+
+                    return assay.hasLookup(sampleModel.queryInfo.schemaQuery);
+                })
+                .sort(naturalSortByProperty('name'))
+                .reduce((_configs, assay) => {
+                    const _queryConfig = createQueryConfigFilteredBySample(
+                        assay,
+                        allSampleIds,
+                        Filter.Types.IN,
+                        (fieldKey, sampleIds) => `${fieldKey} IN (${sampleIds.join(',')})`,
+                        false,
+                        !isSourceSampleAssayGrid
+                    );
+
+                    if (_queryConfig) {
+                        const modelId = `unfiltered-assay-detail:${assay.id}:${sampleId ?? sourceId + '-source'}`;
+                        _configs[modelId] = _queryConfig;
+                    }
+
+                    return _configs;
+                }, {});
+
+            configs = { ...configs, ...unfilteredConfigs };
+        }
+
         return { queryConfigs: configs, tabOrder: _tabOrder };
-    }, [assayModel.definitions, loadingDefinitions, sampleModel]);
+    }, [
+        assayModel.definitions,
+        loadingDefinitions,
+        sampleModel,
+        sampleId,
+        sampleIds,
+        activeSampleAliquotType,
+        showAliquotViewSelector,
+        sourceId,
+        allSampleIds,
+        isSourceSampleAssayGrid,
+    ]);
```

Comment:
This seems like a candidate for refactoring to a method that can have jest tests.